File: sdk/python/meerkat_mobkit/config/session_store.py
"""Session store configuration for MobKit runtime."""
from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class JsonSessionStoreConfig:
    path: str
    stale_lock_threshold_seconds: int = 30

    def to_dict(self) -> dict[str, Any]:
        return {
            "store": "json_file",
            "path": self.path,
            "stale_lock_threshold_seconds": self.stale_lock_threshold_seconds,
        }


@dataclass(frozen=True)
class BigQuerySessionStoreConfig:
    dataset: str
    table: str
    project_id: str | None = None
    gc_interval_hours: int = 6

    def to_dict(self) -> dict[str, Any]:
        result: dict[str, Any] = {
            "store": "bigquery",
            "dataset": self.dataset,
            "table": self.table,
            "gc_interval_hours": self.gc_interval_hours,
        }
        if self.project_id:
            result["project_id"] = self.project_id
        return result


def json(path: str, **kwargs: Any) -> JsonSessionStoreConfig:
    return JsonSessionStoreConfig(path=path, **kwargs)


def bigquery(dataset: str, table: str, **kwargs: Any) -> BigQuerySessionStoreConfig:
    return BigQuerySessionStoreConfig(dataset=dataset, table=table, **kwargs)

File: sdk/typescript/src/config/session-store.ts
/**
 * Session store configuration for MobKit runtime.
 */

// -- JSON file store ------------------------------------------------------

export interface JsonSessionStoreConfig {
  readonly path: string;
  readonly staleLockThresholdSeconds: number;
  toDict(): Record<string, unknown>;
}

export function json(
  path: string,
  options?: { staleLockThresholdSeconds?: number },
): JsonSessionStoreConfig {
  const config = {
    path,
    staleLockThresholdSeconds: options?.staleLockThresholdSeconds ?? 30,
    toDict() {
      return jsonSessionStoreConfigToDict(config);
    },
  };
  return config;
}

export function jsonSessionStoreConfigToDict(
  config: JsonSessionStoreConfig,
): Record<string, unknown> {
  return {
    store: "json_file",
    path: config.path,
    stale_lock_threshold_seconds: config.staleLockThresholdSeconds,
  };
}

// -- BigQuery store -------------------------------------------------------

export interface BigQuerySessionStoreConfig {
  readonly dataset: string;
  readonly table: string;
  readonly projectId: string | null;
  readonly gcIntervalHours: number;
  toDict(): Record<string, unknown>;
}

export function bigquery(
  dataset: string,
  table: string,
  options?: { projectId?: string; gcIntervalHours?: number },
): BigQuerySessionStoreConfig {
  const config = {
    dataset,
    table,
    projectId: options?.projectId ?? null,
    gcIntervalHours: options?.gcIntervalHours ?? 6,
    toDict() {
      return bigquerySessionStoreConfigToDict(config);
    },
  };
  return config;
}

export function bigquerySessionStoreConfigToDict(
  config: BigQuerySessionStoreConfig,
): Record<string, unknown> {
  const result: Record<string, unknown> = {
    store: "bigquery",
    dataset: config.dataset,
    table: config.table,
    gc_interval_hours: config.gcIntervalHours,
  };
  if (config.projectId) result.project_id = config.projectId;
  return result;
}
